lib/index.js

```diff
@@ -538,16 +540,33 @@ api.verify = async ({
     });
   }
 
-  const token = record.meta['bedrock-authn-token'].tokens[type];
-  if(token.expires && new Date() >= new Date(token.expires)) {
-    throw new BedrockError(
-      'Authentication token has expired.',
-      'NotFoundError', {
-        httpStatusCode: 404,
-        public: true
-      });
+  let tokenValue = record.meta['bedrock-authn-token'].tokens[type];
+  if(type === 'nonce') {
+    for(const token of tokenValue) {
+      if(token.sha256 === prefixedHash(hash)) {
+        if(token.expires && new Date() >= new Date(token.expires)) {
+          throw new BedrockError(
+            'Authentication token has expired.',
+            'NotFoundError', {
+              httpStatusCode: 404,
+              public: true
+            });
+        }
+        tokenValue = token;
+      }
+    }
+  } else {
+    if(tokenValue.expires && new Date() >= new Date(tokenValue.expires)) {
+      throw new BedrockError(
+        'Authentication token has expired.',
+        'NotFoundError', {
+          httpStatusCode: 404,
+          public: true
+        });
+    }
```

Comment:
```suggestion
  } else if(tokenValue.expires && now >= tokenValue.expires) {
    throw new BedrockError(
      'Authentication token has expired.',
      'NotFoundError', {
        httpStatusCode: 404,
        public: true
      });
  }
```

Reply:
@dlongley @davidlehn It looks like right now the only token type with expiry date on it is nonce type. So I think that the else if here is not actually being used. Should we remove it?

Reply:
Addressed here e107909